tools/roxvet/analyzers/gogoprotofunctions/analyzer.go

```diff
@@ -0,0 +1,71 @@
+package gogoprotofunctions
+
+import (
+	"fmt"
+	"go/ast"
+	"go/types"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+	"golang.org/x/tools/go/types/typeutil"
+)
+
+const (
+	doc = `check for usages of "github.com/gogo/protobuf/types" functions`
+
+	gogoProtoPkg = "github.com/gogo/protobuf/proto"
+
+	messageFormat = "Cannot call directly the %q function from the \"github.com/gogo/protobuf/types\" package. " +
+		"Use the %q function from the \"github.com/stackrox/rox/pkg/protocompat\" package instead."
+)
+
+var (
+	allowedCallerPackages = []string{
+		"github.com/stackrox/rox/pkg/protocompat",
+		"github.com/stackrox/rox/pkg/protoconv",
+	}
+
+	monitoredFunctions = map[string]string{
+		// Key is the function name in gogo protobuf,
+		// value is the replacement function name in protocompat
+		"Equal": "Equal",
+	}
```

Comment:
Is there a particular reason you have to check for individual function calls? You seem to remove all usages of `github.com/gogo/protobuf/types` from central and sensor through various PRs.

Reply:
The point of this roxvet rule is to ensure that the gogoproto functions do not get called directly again in the codebase by code changes. The proto library change is a big chunk of work and concurrent changes may extend again the code impacts of the proto library change. We'd like to keep the changes contained.